Add two-column mobile-friendly Hugo site.

prideout opened the pull request:
This is a simple place to host WebGL demos and tutorials, we can prettify later. It will show up at:

    https://google.github.io/filament/

The hugo config specifies the output holder to be ../docs which is where our GitHub Pages site is located.

Note that this commit adds the *source* to the site, it does not publish the actual site. When we're ready to publish the actual site, we'll do:

```terminal
cd site ; hugo ; cd ..
git add docs ; git commit
```

![hugosite2](https://user-images.githubusercontent.com/1288904/47756866-fdedee80-dc60-11e8-8ca7-16e02ffe1588.gif)

romainguy commented:
My only comment is that I don't like the theme :) The content is also extremely Filament-JS oriented. We should definitely surface that but the google.github.io website should provide a clear description of the project with a section about the JS binding, not just the JS bindings/tutorials/docs.

prideout commented:
I added content to the landing page and added a live demo there.  I also reorganized the content so that all webgl stuff is on one page, and all "core filament" stuff is on another page.  Screenshots below.

Also changed blue to gray but yeah the theme isn't great.

<img width="1015" alt="landing" src="https://user-images.githubusercontent.com/1288904/47825181-be410880-dd2d-11e8-8411-0194a6aa7653.png">
<img width="563" alt="mobile" src="https://user-images.githubusercontent.com/1288904/47825182-c0a36280-dd2d-11e8-8e36-4ed780786234.png">
<img width="1015" alt="webdocs" src="https://user-images.githubusercontent.com/1288904/47825183-c26d2600-dd2d-11e8-92aa-2119077771f4.png">
<img width="1015" alt="coredocs" src="https://user-images.githubusercontent.com/1288904/47825186-c436e980-dd2d-11e8-9ac8-e7fe0e699f76.png">
